third_party/color_emoji/add_glyphs.py

```diff
@@ -159,6 +172,12 @@ def add_lig_sequence(ligatures, seq, n):
         if 'u200D' in seq:
                 rev_seq = seq[:]
                 rev_seq.reverse()
+                for i in xrange(1, len(rev_seq)):
+                  if is_fitzpatrick(rev_seq[i - 1]):
+                    tmp = rev_seq[i]
+                    rev_seq[i] = rev_seq[i-1]
+                    rev_seq[i-1] = tmp
```

Comment:
FWIW, from my perspective it is unfortunate that in order to handle these
emoji sequences in RTL contexts we need to duplicate the ligature sequences
in the GSUB table.  The ZWJ mechanism used to create these emoji is itself
unfortunate, though, so what can you do....

On Mon, Jul 11, 2016 at 11:46 AM, Behdad Esfahbod <notifications@github.com>
wrote:

> In third_party/color_emoji/add_glyphs.py
> <https://github.com/googlei18n/noto-emoji/pull/59#discussion_r70313940>:
>
> > @@ -159,6 +172,12 @@ def add_lig_sequence(ligatures, seq, n):
> >          if 'u200D' in seq:
> >                  rev_seq = seq[:]
> >                  rev_seq.reverse()
> > +                for i in xrange(1, len(rev_seq)):
> > +                  if is_fitzpatrick(rev_seq[i - 1]):
> > +                    tmp = rev_seq[i]
> > +                    rev_seq[i] = rev_seq[i-1]
> > +                    rev_seq[i-1] = tmp
>
> FWIW, swapping in Python can be done like this as well:
>
> rev_seq[i],rev_seq[i-1] = rev_seq[i-1],rev_seq[i]
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlei18n/noto-emoji/pull/59/files/be1ec4f152443e4eea1eeddae78834354008beb0#r70313940>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJepDHpGoqzpFZ77HbP_rtfuDXuLX_Gbks5qUo-ggaJpZM4JJp1E>
> .
>